ckanext/dcat/profiles.py

```diff
@@ -570,6 +573,36 @@ def _last_catalog_modification(self):
             return result['results'][0]['metadata_modified']
         return None
 
+    def _get_source_catalog(self, dataset_ref):
+        '''
+        Returns Catalog reference that is source for this dataset. 
+
+        Catalog referenced in dct:hasPart is returned, 
+        if dataset is linked there, otherwise main catalog 
+        will be returned.
+
+        This will not be used if ckanext.dcat.expose_subcatalogs
+        configuration option is set to False.
+        '''
+        if not asbool(config.get(DCAT_EXPOSE_SUBCATALOGS)):
+            return
+        cats = set(self.g.subjects(DCAT.dataset, dataset_ref))
```

Comment:
Let's be explicit with variable names to avoid feline confusions and call this `catalogs` :)